core/src/main/javascript/io/Url.js

```diff
@@ -269,16 +270,25 @@
          * return undefined until read() is first called, and may still return
          * undefined afterwards if the JSON cannot be accessed in this
          * manner.</p>
-         * 
+         *
          * <p>If read() does result in JSON becoming available via this method,
          * then read() will return zero bytes of data.</p>
-         * 
+         *
          * @return {?Array{*}} an array of JSON values or undefined.
          */
         getJSON: function getJSON() {
             return this._json;
         },
 
+        /**
+        * <p>Returns an array of HTTP response headers, or undefined.</p>
+        *
+        * @return {?Array{string}} an array of HTTP response headers or undefined.
```

Comment:
Returning an empty array is safer and more consistent with the rest of the code base. `_headers` can be initialized to an empty array.

`getJSON()` is special because `undefined` implies unsupported, as opposed to supported but a value of null.